go/cmd/betabot/main.go

```diff
@@ -103,15 +101,71 @@ func handleEvent(ctx context.Context, messengerClient bertymessenger.MessengerSe
 			if err != nil {
 				return err
 			}
+		} else if contact.State == bertymessenger.Contact_Established {
+			// When contact was established, send message and a group invitation
+			time.Sleep(5 * time.Second)
```

Comment:
I think you should use time.After here

Reply:
I don't think, `time.After` is way costier than `Sleep` and this code is running in a goroutine so no need of it.